upip/upip_utarfile.py

```diff
@@ -39,7 +39,13 @@ def readinto(self, buf):
         return sz
 
     def skip(self):
-        self.f.read(self.content_len + self.align)
+        sz = self.content_len + self.align
+        if sz:
+            buf = bytearray(16)
+            while sz:
+                s = min(sz, 16)
+                self.f.readinto(buf, s)
+                sz -= s
```

Comment:
DecompIO could only support SEEK_CUR with a non-negative argument, and the return value would have to be 0 I'd say (basically you're always at the start of the stream).  That's essentially a skip(n) method.  I think we can leave it for now but remember that if a use for skipping comes up again then we can implement it.